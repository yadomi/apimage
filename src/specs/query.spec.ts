const { parse, validate } = require('../helpers/query');

it('should parse query string to object', () => {
  expect(parse('w_100', 'jpeg')).toMatchObject({
    width: '100',
    extension: 'jpeg',
  });

  expect(parse('w_100,h_200,f_cover,q_100,p_west', 'jpeg')).toMatchObject({
    width: '100',
    height: '200',
    fit: 'cover',
    quality: '100',
    position: 'west',
    extension: 'jpeg',
  });

  expect(parse('w_100,h_200,f_cover,q_100,p_west', 'jpeg')).toMatchSnapshot();
});

it('should return an error when validating invalid query', () => {
  const parsed = parse('w_4000', 'jpeg');
  const result = validate(parsed);
  expect(result).toHaveProperty('error');
  expect(result).toHaveProperty('error.name', 'ValidationError');
});

it('should convert string to number for specific properties', () => {
  const parsed = parse('w_100', 'jpeg');
  const result = validate(parsed);
  expect(result.value).toMatchObject({ width: 100, extension: 'jpeg' });
  expect(typeof result.value.width).toBe('number');
  expect(result.value).toMatchSnapshot();
});
