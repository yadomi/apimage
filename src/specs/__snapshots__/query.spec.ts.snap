// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert string to number for specific properties 1`] = `
Object {
  "extension": "jpeg",
  "width": 100,
}
`;

exports[`should parse query string to object 1`] = `
Object {
  "extension": "jpeg",
  "fit": "cover",
  "height": "200",
  "position": "west",
  "quality": "100",
  "width": "100",
}
`;
